### PR TITLE
Fix duplicate and inconsistent rendering sequence

### PR DESCRIPTION
**Pull request by petyosi**
One of the issues found with the problem is described in #97.
Probably many other reincarnations exist. 

The root cause of the issue was that prop updates (in this case, the `renderItem` prop) 
had higher priority compared to the observable updates. 
The solution was to move the render function calculation inside the store.

### CODE REVIEW

File: src/Virtuoso.tsx [hunk 1]
```diff
@@ -98,7 +87,7 @@ export interface VirtuosoMethods {
 }
 
 export const Virtuoso = forwardRef<VirtuosoMethods, VirtuosoProps>((props, ref) => {
-  const [state] = useState(VirtuosoStore(props))
+  const [state] = useState(() => VirtuosoStore(props))
```
Comment (petyosi) [inline src/Virtuoso.tsx:90]: @toonvd - this was the reason for the empty list you were observing. I am the noob who does not know how to use hooks in this case; With each re-render a new store instance was created, only to be immediately discarded. These are the specifics of the `useState` hook.

However, this was not the real problem. You can review the rest of the changes - and if you wish, please test the branch on your side.

Comment (toonvd) [reply src/Virtuoso.tsx:91]: Since we both come from classes, I think this was pretty hard to detect xD I have a few hours tommorow for thorough testing!

Comment (petyosi) [reply src/Virtuoso.tsx:91]: Can you please comment here when done? If not, I am publishing this on Monday, collected a few more from @lesha1201 in this release.

Comment (toonvd) [reply src/Virtuoso.tsx:91]: Absolutely. I'll try to do it in the morning!

Comment (toonvd) [reply src/Virtuoso.tsx:91]: Hi @petyosi, first tests with playground are going well. I notice a performance increase and no more errors. #91 seems to be fixed for me also. Is there any way to get this branch in one of my projects to doublecheck?

Comment (petyosi) [reply src/Virtuoso.tsx:91]: Yes @toonvd . Run `yarn build`, and `yarn pack`. You will get a tarball file in the root dir. After that, from your project, run `npm i ../path/to/virtuoso/tarball.tgz` (fix the exact file name).

Comment (toonvd) [reply src/Virtuoso.tsx:91]: Hi, selenium tests passed and some manual tests also on that particular project.
I think this should be ok for a new version.

**Pull request merged**